server/elasticsearch/src/test/java/org/hawkular/apm/server/elasticsearch/TraceServiceElasticsearchTest.java

```diff
@@ -507,4 +432,102 @@ public void testStoreAndRetrieveComplexTraceById() {
                 .get(0).getUri());
     }
 
+    @Test
+    public void testStoreAndRetrieveCausedByTraceById() throws StoreException, JsonProcessingException {
+        Trace trace1 = new Trace();
+        trace1.setId("1");
+        trace1.setStartTime(System.currentTimeMillis());
+        Consumer c1 = new Consumer();
+        c1.setUri("uri1");
+        c1.getProperties().add(new Property("prop1","value1"));
+        trace1.getNodes().add(c1);
+        Component comp1 = new Component();
+        comp1.setUri("comp1");
+        c1.getNodes().add(comp1);
+
+        Trace trace2 = new Trace();
+        trace2.setId("2");
+        trace2.setStartTime(System.currentTimeMillis());
+        Consumer c2 = new Consumer();
+        c2.setUri("uri2");
+        c2.getProperties().add(new Property("prop1","value1"));
+        c2.getProperties().add(new Property("prop2","value2"));
+        c2.addCausedByCorrelationId(trace1.getId()+":0:0");
+        trace2.getNodes().add(c2);
+        Component comp2 = new Component();
+        comp2.setUri("comp2");
+        c2.getNodes().add(comp2);
+
+        Trace trace3 = new Trace();
+        trace3.setId("3");
+        trace3.setStartTime(System.currentTimeMillis());
+        Consumer c3 = new Consumer();
+        c3.setUri("uri3");
+        c3.getProperties().add(new Property("prop3","value3"));
+        c3.addCausedByCorrelationId(trace1.getId()+":0:0");
+        trace3.getNodes().add(c3);
+        Component comp3 = new Component();
+        comp3.setUri("comp3");
+        c3.getNodes().add(comp3);
+
+        ts.storeFragments(null, Arrays.asList(trace1, trace2, trace3));
+
+        // Retrieve stored trace
+        Wait.until(() -> ts.getTrace(null, "1") != null);
+        Trace result = ts.getTrace(null, "1");
+
+        assertNotNull(result);
+        assertEquals("1", result.getId());
+
+        ObjectMapper mapper = new ObjectMapper();
+        mapper.enable(SerializationFeature.INDENT_OUTPUT);
+        System.out.println("TRACE(testStoreAndRetrieveSingleCausedByTraceById)=" + mapper.writeValueAsString(result));
+
+        assertEquals(3, result.allProperties().size());
+        assertEquals(1, result.getNodes().size());
```

Comment:
Prefer assertEquals as it directly reports what the inequality is - rather than just a boolean result